File: 1046-last-stone-weight/1046-last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        int x , y ;
        sort(stones.begin(),stones.end());
        while(stones.size() > 1){
            int m  = stones.size();
            x =  stones[m-2];
            y = stones[m-1];
            if(x==y){
                stones.pop_back();
                stones.pop_back();
            }
            else{
                stones.pop_back();
                stones.pop_back();
                stones.push_back(y-x);
                sort(stones.begin(),stones.end());
            }
        }
        if (stones.empty()) return 0;
        return stones[0];
        
    }
};

File: 1046-last-stone-weight/1046-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        res  = sorted(stones,reverse = True)
        
        while len(res) > 1 :
            x =  res[1] 
            y = res[0]
            #print(res)
            if x ==  y : 
                res = res[2:]
            else :
                s =  y -x 
                res = res[2:]
                res.append(s)
                res.sort(reverse = True)
        if not res : 
            return 0
        return res[0]
           